ia32/plo.c

```diff
@@ -213,34 +213,52 @@ void plo_unlimitfs(void)
 	low_setfar(SYSPAGE_SEG, SYSPAGE_OFFS_GDTR + 2, GDT_SEG << 4);
 	low_setfar(SYSPAGE_SEG, SYSPAGE_OFFS_GDTR + 4, GDT_SEG >> 12);
 
-	/* Disable interrupts */
 	low_cli();
 
 #asm
-	push bx
+	; Save real mode segment registers
+	push ds
+	push es
+	push fs
+	push gs
+
+	; Load GDT into GDTR
+	mov ax, #SYSPAGE_SEG
+	mov es, ax
+	seg es
+	lgdt 0
 
-	; Load gdt register
-	mov eax, #SYSPAGE_SEG
-	lgdt [eax]
-	
 	; Switch to pmode by setting pmode bit
 	mov eax, cr0
-	or al, 1
+	inc eax
```

Comment:
Are we 100% sure LSB is not set already in this case?

Reply:
Yes, plo starts in the real mode with cr0 LSB zeroed out. Confirmed by Paweł.